pkg/services/accesscontrol/accesscontrol.go

```diff
@@ -49,3 +50,16 @@ func BuildPermissionsMap(permissions []*Permission) map[string]bool {
 
 	return permissionsMap
 }
+
+// GroupPermissions will group scopes on action
+func GroupPermissions(permissions []*Permission) map[string]map[string]struct{} {
```

Comment:
No strong opinion there, but wdyt of:
```suggestion
func GroupPermissionsByAction(permissions []*Permission) map[string]map[string]struct{} {
```